src/fn/upload-app-settings.js

```diff
@@ -39,7 +39,7 @@ async function augmentDeclarativeWithNoolsBoilerplate(appSettings) {
   }
 
   const actualCoreVersion = await getValidApiVersion(appSettings);
-  const addNoolsBoilerplate = semver.lt(actualCoreVersion, '4.2.0-dev');
+  const addNoolsBoilerplate = actualCoreVersion && semver.lt(actualCoreVersion, '4.2.0-dev');
```

Comment:
I'm unclear on when `actualCoreVersion` would be falsy... Why?

Reply:
Even though coerce tries to parse versions correctly, it'd be falsy when a custom version like tag number or branch number is being used on production. What do you suggest in this case to avoid scenarios like you mentioned? One option is to use `base_Version` which is only available in latest CHT versions and ensure this doesn't get changed. This issue was also caused because the way version was written was changed.

Reply:
Can you share the specific tags you're having trouble with please? I'm testing with coerce and not clear on what tags are failing.

Reply:
This issue fix was needed for regular CHT versions as well like `4.5.0.6922454971`, hence I introduced coerce for that fix. `actualCoreVersion` could still be falsy if production instances are running on versions like [this](https://github.com/medic/config-nssd/issues/65) (private). We won't need falsy check after introducing coerce. I don't see recent instances with unusual versioning though. I am not sure what `coreVersion` would've returned for particular versions like that since `api/deploy-info` endpoint has changed frequently.